staging/src/k8s.io/client-go/tools/leaderelection/leaderelection_test.go

```diff
@@ -302,6 +317,9 @@ func testTryAcquireOrRenew(t *testing.T, objectType string) {
 			}
 
 			le.maybeReportTransition()
+			if test.disableLockCreation {
+				wg.Done()
```

Comment:
Maybe also add a 30s sleep? We want to make sure no one becomes the leader asynchronously. We can sleep and verify `reportedLeader` is still emtpy.

Reply:
I don't think that anyone can become the leader asynchronously if `disableLockCreation` is `true`. This can be verified by the fact that if we remove this `if` block, the test will hang since the `OnNewLeader` callback which calls `wg.Done()` will never be called. Is it necessary to have a fairly large sleep period to test something which is redundant(imo, please correct me if i'm wrong here) ?